tee-worker/ts-tests/integration-tests/common/utils/assertion.ts

```diff
@@ -244,64 +244,71 @@ export async function assertIdGraphMutationResult(
 
 export async function assertVc(context: IntegrationTestContext, subject: CorePrimitivesIdentity, data: Bytes) {
     const results = context.api.createType('RequestVCResult', data) as unknown as RequestVCResult;
-    const vcHash = results.vc_hash.toString();
-
     // step 1
-    const vcIndex = results.vc_index.toString();
-    const vcRegistry = (await context.api.query.vcManagement.vcRegistry(
-        vcIndex
-    )) as unknown as PalletVcManagementVcContext;
-    const vcStatus = vcRegistry.toHuman()['status'];
-    assert.equal(vcStatus, 'Active', 'Check VcRegistry error:status should be equal to Active');
-
-    // step 2
     // decryptWithAes function added 0x prefix
     const vcPayload = results.vc_payload;
     const decryptVcPayload = decryptWithAes(aesKey, vcPayload, 'utf-8').replace('0x', '');
 
-    const vcPayloadHash = blake2AsHex(Buffer.from(decryptVcPayload));
-    assert.equal(vcPayloadHash, vcHash, 'Check VcPayload error: vcPayloadHash should be equal to vcHash');
-
     /* DID format
     did:litentry:substrate:0x12345...
     did:litentry:evm:0x123456...
     did:litentry:twitter:my_twitter_handle
     */
 
-    // step 3
+    // step 2
+    // check credential subject's DID
     const credentialSubjectId = JSON.parse(decryptVcPayload).credentialSubject.id;
     const expectSubject = Object.entries(JSON.parse(subject.toString()));
 
+    // step 3
     // convert to DID format
     const expectDid = 'did:litentry:' + expectSubject[0][0] + ':' + expectSubject[0][1];
     assert.equal(
         expectDid,
         credentialSubjectId,
-        'Check credentialSubjec error: expectDid should be equal to credentialSubject id'
+        'Check credentialSubject error: expectDid should be equal to credentialSubject id'
     );
 
     // step 4
+    // extrac proof and vc without proof json
     const vcPayloadJson = JSON.parse(decryptVcPayload);
     const { proof, ...vcWithoutProof } = vcPayloadJson;
-    assert.equal(vcIndex, vcPayloadJson.id, 'Check VcIndex error: VcIndex should be equal to vcPayload id');
 
     // step 5
+    // prepare teerex enclave registry data for further checks
     const enclaveCount = await context.api.query.teerex.enclaveCount();
+    const parachainBlockHash = await context.api.query.system.blockHash(vcPayloadJson.parachainBlockNumber);
+    const apiAtVcIssuedBlock = await context.api.at(parachainBlockHash);
+    const lastRegisteredEnclave = (await apiAtVcIssuedBlock.query.teerex.enclaveRegistry(enclaveCount))
+        .value as TeerexPrimitivesEnclave;
```

Comment:
I think checking `scheduledEnclave` should be more accurate - enclave registration needs to comply with it too. Right now I can't think of any case where `scheduledEnclave` and `enclaveRegistry` don't match though

Btw even if we want to check `enclaveRegistry` we should `enclaveCount` from the `parachainBlockHash` height

Reply:
But Teerex's `ScheduledEnclave` is empty so there is no value to compare against :(

Reply:
Ah in `dev` it's not populated, but in production it shouldn't be empty

Reply:
Maybe we should add the scheduled enclave even in `dev` - like here: https://github.com/litentry/litentry-parachain/pull/2382/files#diff-f941434353cbdda40f528800d59a89b42e5e51396b03c092bf41ef8e294d8fd5

Reply:
Maybe we can always populate it with workers mrenclave for first validateer in dev see: https://github.com/litentry/litentry-parachain/pull/2382#issuecomment-1892237305

Reply:
@Kailai-Wang, @felixfaisal  what do you think?

Reply:
@Kailai-Wang @kziemianek I think we can always populate it for first validateer.